src/vmspawn/vmspawn.c

```diff
@@ -2229,8 +2239,10 @@ static int verify_arguments(void) {
         if (!strv_isempty(arg_initrds) && !arg_linux)
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Option --initrd= cannot be used without --linux=.");
 
-        if (arg_register && !arg_privileged)
-                return log_error_errno(SYNTHETIC_ERRNO(EPERM), "--register= requires root privileges, refusing.");
+        if (arg_keep_unit && arg_register && cg_pid_get_owner_uid(0, NULL) >= 0)
```

Comment:
I don't think the check is enough? If vmspawn is running as a user service, which in turn has a user slice unit, we would accept it and `machine_stop` would eventually kill `user@.service`. We need to check this properly in server side and tighten the check on client side.